src/org/zaproxy/zap/db/sql/SqlTableHistory.java

```diff
@@ -341,89 +341,112 @@ private RecordHistory build(ResultSet rs) throws HttpMalformedHeaderException, D
 		}
 	
 	}
+
+	private List<Integer> getHistoryIdsFromPreparedStatement(PreparedStatementFactory preparedStatementFactory) throws DatabaseException {
+		SqlPreparedStatementWrapper psGetHistoryIds = null;
+		try {
+
+			psGetHistoryIds = preparedStatementFactory.create();

```

Comment:
This has the side effect of leaking the prepared statement if an exception occurs while setting the parameters, better keep the creation in this method (also less code duplication) and let the function just set the parameters.